src/planner/binder/statement/bind_create.cpp

```diff
@@ -112,6 +114,27 @@ SchemaCatalogEntry *Binder::BindCreateFunctionInfo(CreateInfo &info) {
 	return BindSchema(info);
 }
 
+void Binder::BindLogicalType(ClientContext &context, LogicalType &type, const string &schema) {
+	if (type.id() == LogicalTypeId::LIST) {
+		auto &child_type = ListType::GetChildTypeBind(type);
+		BindLogicalType(context, child_type, schema);
+	} else if (type.id() == LogicalTypeId::STRUCT || type.id() == LogicalTypeId::MAP) {
+		auto &child_types = StructType::GetChildTypesBind(type);
+		for (auto &child_type : child_types) {
+			BindLogicalType(context, child_type.second, schema);
+		}
+	} else if (type.id() == LogicalTypeId::USER) {
+		auto &user_type_name = UserType::GetTypeName(type);
+		auto user_type_catalog = (TypeCatalogEntry *)context.db->GetCatalog().GetEntry(context, CatalogType::TYPE_ENTRY,
+		                                                                               schema, user_type_name, true);
+		if (!user_type_catalog) {
+			throw NotImplementedException("DataType %s not supported yet...\n", user_type_name);
+		}
+		// FIXME: Avoid copy in the future
```

Comment:
The code as-is is fine, the only part that will be copied is what is in the LogicalType which is very small (16 bytes?).